### PR TITLE
Feature signed requests

### PR DESCRIPTION
**Pull request by fgodino**
Hi Aaron, I added a new feature to protect transformations and resources via url signatures. What do you think?

### CODE REVIEW

File: lib/security/security-defaults.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  enabled : false,
+  secret : 'keyboard_cat',
```
Comment (asilvas) [inline lib/security/security-defaults.js]: For a default, I would set this to `null` or `undefined`, and throw an error at initialization of `security.enabled is true but `security.secret` is falsey to force users of this feature to provide an explicit secret.

File: README.md [hunk 2]
```diff
@@ -244,24 +245,53 @@ Most router defaults should suffice, but you have full control over routing. See
...
-* `router.paramValDelimiter` (default: `":"`) - Seperator between a parameter key and its value.
+* `router.cmdKeyDelimiter` (default: `"/"`) - Separator between commands (aka image steps).
+* `router.cmdValDelimiter` (default: `"="`) - Separator between a command and its parameters.
+* `router.paramKeyDelimiter` (default: `","`) - Separator between command parameters.
+* `router.paramValDelimiter` (default: `":"`) - Separator between a parameter key and its value.
+* `router.signatureDelimiter` (default: `"/-/"`) - Separator between steps and the signed url
 * `router.originalSteps` (default: [Full Defaults](https://github.com/asilvas/node-image-steam/blob/master/lib/router/router-defaults.js)) - Steps performed on the original asset to optimize subsequent image processing operations. This can greatly improve the user experience for very large, uncompressed, or poorly compressed images.
 * `router.steps` (default: [Full Defaults](https://github.com/asilvas/node-image-steam/blob/master/lib/router/router-defaults.js)) - Mapping of URI image step commands and their parameters. This allows you to be as verbose or laconic as desired.
 
+## Security Options
```
Comment (asilvas) [inline README.md]: Great feature! But please move this section to bottom right before `# Things to try`

File: README.md [hunk 3]
```diff
@@ -244,24 +245,53 @@ Most router defaults should suffice, but you have full control over routing. See
...
+  "security": {
+    "enabled": false,
+    "secret": keyboard_cat,
+    "algorithm": 'sha1'
+  }
+}
+```
+
+* `security.enabled` (default: `false`) - Security enabled.
+* `security.secret` (default: `keyboard_cat`) - The signing secret
```
Comment (asilvas) [inline README.md]: Please remove default per comment below

File: README.md [hunk 4]
```diff
@@ -244,24 +245,53 @@ Most router defaults should suffice, but you have full control over routing. See
...
+```
 
 # Routing
 
 Routing is left-to-right for legibility.
 
 Routing format:
 
-  `{path}{pathDelimiter}{cmd1}{cmdValDelimiter}{cmd1Param1Key}{paramValDelimiter}{cmd1Param1Value}{paramKeyDelimiter}{cmdKeyDelimiter}?{queryString}`
+  `{path}{pathDelimiter}{cmd1}{cmdValDelimiter}{cmd1Param1Key}{paramValDelimiter}{cmd1Param1Value}{paramKeyDelimiter}{cmdKeyDelimiter}/{signatureDelimiter}/{signature}?{queryString}`
```
Comment (asilvas) [inline README.md]: `/{signatureDelimiter}/` should be `{signatureDelimiter}` as the forward slashes are part of the delimiter.

File: lib/router/router.js [hunk 5]
```diff
@@ -89,14 +89,19 @@ p.getInfo = function(req) {
...
+
   // encoding does not belong here -- rely on storage providers to encode as necessary
   routeInfo.urlInfo.pathname = decodeURI(routeInfo.urlInfo.pathname);
   routeInfo.isCachable = !this.canDisableCache // cannot disable cache
     || routeInfo.urlInfo.query.cache !== 'false' // or request isn't disabling cache
   ;
 
   // break apart imagePath from imageSteps from queryParams
-  var pathParts = routeInfo.urlInfo.pathname.split(this.options.pathDelimiter);
+  var pathParts = routeInfo.urlInfo.pathname.split(this.options.signatureDelimiter)[0].split(this.options.pathDelimiter);
```
Comment (asilvas) [inline lib/router/router.js]: I don't see why this change would be needed. Please explain.

File: scripts/dev.js [hunk 6]
```diff
@@ -23,5 +23,9 @@ module.exports = {
     cache: {
       path: path.resolve(__dirname, '../test/cache')
     }
+  },
+  security: {
+    enabled : false,
+    secret : 'keyboard_cat'
```
Comment (asilvas) [inline scripts/dev.js]: please remove secret to force explicit value

**Comment by asilvas** [comment]
Thanks, published as `0.34.0`

**Comment by fgodino** [comment]
My pleasure :)
